README file

AlyaEngineer opened the pull request:
## Summary

Added `README.md` file, describing the project, its purpose, and the technology stack used, including available scripts and  instructions for setting up and running the project locally

Task in Jira: [ECA-19](https://oiki.atlassian.net/browse/ECA-19), [ECA-20](https://oiki.atlassian.net/browse/ECA-20), [ECA-21](https://oiki.atlassian.net/browse/ECA-21)
Task in RS Shool: [ECOMM-1_17](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_17.md), [RSS-ECOMM-1_19](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_19.md), [RSS-ECOMM-1_20](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_20.md)
[Link to discussion](https://discord.com/channels/1363571469237489964/1363571471414202539/1365758071963652187)

## Checklist

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

<!-- Optional Sections -->
<details>
<summary><strong>Expand for optional sections</strong></summary>

## Screenshots
![Снимок экрана 2025-05-01 130438](https://github.com/user-attachments/assets/16740434-2398-4bc7-a294-befddab2acc0)
![Снимок экрана 2025-05-01 130450](https://github.com/user-attachments/assets/6956b3f2-6923-4880-a7a6-4596da990213)
![Снимок экрана 2025-05-01 130501](https://github.com/user-attachments/assets/cce2687b-5d30-432c-b34b-abdfe1286007)
![Снимок экрана 2025-05-01 130514](https://github.com/user-attachments/assets/007994e7-b133-41a3-97c6-1e2d65e73fc4)

## Special notes for your reviewer
Available scripts, logo and app name will be specified soon


</details>
<!-- End of Optional Sections -->

[ECA-19]: https://oiki.atlassian.net/browse/ECA-19?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECA-20]: https://oiki.atlassian.net/browse/ECA-20?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ECA-21]: https://oiki.atlassian.net/browse/ECA-21?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AlyaEngineer committed:
docs: (ECA-19) add README.md file in the repository

describe the project, its purpose and the technology stack

AlyaEngineer committed:
docs: (ECA-20) add scripts to README file

AlyaEngineer committed:
docs: (ECA-21) add instructions to README file for setting up and running the project locally

AlyaEngineer committed:
docs: (ECA-19) add link for deploy shield, add links to the other shields

Mxmmsv commented on README.md:
````diff
@@ -1 +1,106 @@
+In the project directory, you can run:
+
+```
+$ npm run start
````
```suggestion
$ npm run dev
```
В `Vite` по-умолчанию используется такой скрипт. Можем оставить и `npm run start`, но придётся тогда редактировать `package.json`

ek-ole replied:
Я за `npm run dev` потому что уже привыкла так

Mxmmsv commented on README.md:
````diff
@@ -1 +1,106 @@
+# Runs tests in watch mode.
+```
+
+You can also use `yarn` or `pnpm` instead of `npm`, depending on your package manager.
````
```suggestion
In the project directory, you can run:

- Runs the app in development mode. Open port 5173 (<http://localhost:5173>) to view the app in the browser.

    ```bash
    npm run dev
    ```

- Builds the app for production to the `dist/` folder

    ```bash
    npm run build
    ```

- Lints all `.ts` and `.tsx` files using ESLint

    ```bash
    npm run lint
    ```

- Formats code using Prettier

    ```bash
    npm run format
    ```

- Runs unit tests using Vitest

    ```bash
    npm run test
    ```

- Runs tests in watch mode

    ```bash
    npm run test:watch
    ```

You can also use `yarn` or `pnpm` instead of `npm`, depending on your package manager.
```

Может быть будем использовать такую структуру? Пользователь сможет сразу скопировать команду, кликнув по ней

![image](https://github.com/user-attachments/assets/f74b5243-12bc-4a8d-adf3-a10cf95473bb)

ek-ole replied:
Good idea!

Mxmmsv commented on README.md:
````diff
@@ -1 +1,106 @@
+- start the local server using `npm run start`:
+```bash
+$ npm run start
+```
````
```suggestion
- Using terminal to go to the directory in which you want to install the application, for example:

    ```bash
    cd C/Users/yourname/apps
    ```

- Clone the repository from GitHub:

    ```bash
    git clone https://github.com/Mxmmsv/eCommerce-Application.git
    ```

- install the dependencies with `npm i`:

    ```bash
    npm i
    ```

- start the local server using `npm run start`:

    ```bash
    npm run start
    ```
```

Добавил отступы для лучшей читабильности, теперь каждая команда относится к конкретному пункту

![image](https://github.com/user-attachments/assets/d8ab702b-d60b-437d-be8f-3f338584332b)

Mxmmsv replied:
Кстати, скорее всего нужно будет добавить ещё один пункт после клонирования репо

- Go to the project directory:

    ```bash
    cd eCommerce-Application
    ```

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+# <h1 align="center">🖼️ PosterShop</h1>
+
+
+## Description
```
```suggestion
## 📝Description
```
Может, добавим эмодзи для разделов?)

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+- User authentication via SDK: Users can sign in securely using SDK for authentication.
+
+
+## Install and Run the Application
```
```suggestion
## 💻⚙️Install and Run the Application
```

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+The server runs on port `5173`. If it is not possible to run the server using port `5173`, you can use another port by modifying the `vite.config.ts` configuration file.
+
+
+## Available Scripts
```
```suggestion
## 📜Available Scripts
```

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+You can also use `yarn` or `pnpm` instead of `npm`, depending on your package manager.
+
+
+## Authors
```
```suggestion
##  👥Authors
```

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+[Maxim Moiseev](https://github.com/Mxmmsv), [Ekaterina Dmitrenko](https://github.com/ek-ole) and [Alla Tsaiukova](https://github.com/AlyaEngineer).
+
+
+## Acknowledgments
```
```suggestion
##  🙏Acknowledgments
```

Mxmmsv commented on README.md:
```diff
@@ -1 +1,106 @@
+Special thanks to the Commercetools team for providing a well-documented, TypeScript-ready SDK that simplified our e-commerce integration.
+
+
+## License
```
```suggestion
## 📄🔒License
```

Mxmmsv reviewed (changes requested):
Очень классно получилось!
![image](https://github.com/user-attachments/assets/1d6f8186-ada2-4513-9b10-3c611ff5d325)

ek-ole reviewed (approved):
Я думаю что эмодзи можно добавить, раз Максим их уже все написал. Так наш Readme будет веселым 😺 
Отличная работа! Очень красиво!

AlyaEngineer committed:
docs: update README file with current scripts

AlyaEngineer committed:
refactor: change the badge for ui kit, add emoji to section titles

change the badge from Material UI to shadcn/ui in accordance with the team's decision, change the logo of the SWR badge

AlyaEngineer committed:
fix: correct the sequence for installing and running the application, improve it formatting

AlyaEngineer committed:
Merge branch 'sprint-1' into docs/add-readme-file
